rust/src/client.rs

```diff
@@ -290,4 +290,12 @@ mod tests {
             running_in_container()
         );
     }
+
+    #[test]
+    fn test_is_bare_split_xattrs() {
+        assert_eq!(
+            Path::new("/ostree/repo/").exists(),
```

Comment:
This test is not going to reliably work because it depends on external global system state.

But I think the real problem is the `is_bare_split_xattrs()` API.
I think it should be tweaked to 1) take an open FD as input argument, and 2) return a `Result<String>`. 
That way it can be properly tested (directly in that repo, not here) and it gives us a bit more flexibility.

Reply:
I am not sure I follow, I think that is for @cgwalters to answer. Why passing a file descriptor would make it any different? Wouldn't the system the test is running on will be a ostree enabled FS or not? We would get false or true depending on where the test is ran.

Reply:
> But I think the real problem is the is_bare_split_xattrs() API.
> I think it should be tweaked to 1) take an open FD as input argument

I was thinking of this API was a short-term hack until we teach the ostree core about this.  We don't at the moment have a use case for introspecting something other than the currently booted root.

> , and 2) return a Result<String>.

Hmm, where the returned string is the repo mode?

Reply:
> This test is not going to reliably work because it depends on external global system state.

This though is true.  I think we can just drop the unit test here - we have coverage for the logic from the ostree-ext tests.

Reply:
Ack, let's just drop the test here then. The FD argument would be just to make testing here easier, but indeed there is no need at all.

Yes, the String would hold the repo mode, as this custom one can't be stored/extracted by the usual C logic.

Reply:
test removed.